Failing tests of `handlerfuncs_test.go` when randomizing execution order of tests

Depending on the root cause(s), this might be a single issue or multiple.
When randomizing the execution order of tests, the following tests of `handlerfuncs_test.go` fail:

```
--- FAIL: TestLoadSchemaHandlerWithReadBodyError (0.00s)
    handlerfuncs_test.go:406: 
                Error Trace:    handlerfuncs_test.go:406
                Error:          "" does not contain "error reading"
                Test:           TestLoadSchemaHandlerWithReadBodyError
2022-06-22T15:34:51.064Z, INFO, defra.http, Request, {"Method": "POST", "Path": "/api/v0/graphql", "Status": 400, "Length": 1387, "Elapsed": 0.000039524}
2022-06-22T15:34:51.064Z, ERROR, defra.http, Error while encoding JSON: json: unsupported value: +Inf
2022-06-22T15:34:51.064Z, ERROR, defra.http, Internal Server Error, {"Error": "no database available", "ErrorVerbose": "no database available\ngithub.com/sourcenetwork/defradb/api/http.dbFromContext\n\t/home/o/dev/defradb/api/http/handler.go:109\ngithub.com/sourcenetwork/defradb/api/http.dumpHandler\n\t/home/o/dev/defradb/api/http/handlerfuncs.go:54\ngithub.com/sourcenetwork/defradb/api/http.(*handler).handle.func1\n\t/home/o/dev/defradb/api/http/handler.go:74\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2047\ngithub.com/go-chi/chi/v5.(*Mux).routeHTTP\n\t/home/o/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:442\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2047\ngithub.com/sourcenetwork/defradb/api/http.loggerMiddleware.func1\n\t/home/o/dev/defradb/api/http/logger.go:57\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2047\ngithub.com/go-chi/chi/v5.(*Mux).ServeHTTP\n\t/home/o/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:88\ngithub.com/sourcenetwork/defradb/api/http.testRequest\n\t/home/o/dev/defradb/api/http/handlerfuncs_test.go:690\ngithub.com/sourcenetwork/defradb/api/http.TestDumpHandlerWithDBError\n\t/home/o/dev/defradb/api/http/handlerfuncs_test.go:119\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1259\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581"}
2022-06-22T15:34:51.064Z, INFO, defra.http, Request, {"Method": "GET", "Path": "/api/v0/debug/dump", "Status": 500, "Length": 1395, "Elapsed": 0.000107693}
2022-06-22T15:34:51.064Z, ERROR, defra.http, Internal Server Error, {"Error": "no database available", "ErrorVerbose": "no database available\ngithub.com/sourcenetwork/defradb/api/http.dbFromContext\n\t/home/o/dev/defradb/api/http/handler.go:109\ngithub.com/sourcenetwork/defradb/api/http.loadSchemaHandler\n\t/home/o/dev/defradb/api/http/handlerfuncs.go:137\ngithub.com/sourcenetwork/defradb/api/http.(*handler).handle.func1\n\t/home/o/dev/defradb/api/http/handler.go:74\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2047\ngithub.com/go-chi/chi/v5.(*Mux).routeHTTP\n\t/home/o/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:442\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2047\ngithub.com/sourcenetwork/defradb/api/http.loggerMiddleware.func1\n\t/home/o/dev/defradb/api/http/logger.go:57\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2047\ngithub.com/go-chi/chi/v5.(*Mux).ServeHTTP\n\t/home/o/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:88\ngithub.com/sourcenetwork/defradb/api/http.testRequest\n\t/home/o/dev/defradb/api/http/handlerfuncs_test.go:690\ngithub.com/sourcenetwork/defradb/api/http.TestLoadSchemaHandlerWithoutDB\n\t/home/o/dev/defradb/api/http/handlerfuncs_test.go:424\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1259\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581"}
```

```
2022-06-22T10:43:22.567-0500, [34mINFO[0m, defra.http, Request, {"Method": "POST", "Path": "/api/v0/graphql", "Status": 400, "Length": 101, "Elapsed": 0.000250666}
--- FAIL: TestExecGQLWithNilBody (0.00s)
    handlerfuncs_test.go:146: 
        	Error Trace:	handlerfuncs_test.go:146
        	Error:      	"" does not contain "body cannot be empty"
        	Test:       	TestExecGQLWithNilBody
2022-06-22T10:43:22.567-0500, [34mINFO[0m, defra.http, Request, {"Method": "POST", "Path": "/api/v0/schema/load", "Status": 400, "Length": 94, "Elapsed": 0.000154625}
--- FAIL: TestLoadSchemaHandlerWithReadBodyError (0.00s)
    handlerfuncs_test.go:406: 
        	Error Trace:	handlerfuncs_test.go:406
        	Error:      	"" does not contain "error reading"
        	Test:       	TestLoadSchemaHandlerWithReadBodyError
2022-06-22T10:43:22.567-0500, [34mINFO[0m, defra.http, Request, {"Method": "POST", "Path": "/api/v0/graphql", "Status": 400, "Length": 94, "Elapsed": 0.000054209}
--- FAIL: TestExecGQLWithMockBody (0.00s)
    handlerfuncs_test.go:195: 
        	Error Trace:	handlerfuncs_test.go:195
        	Error:      	"" does not contain "error reading"
        	Test:       	TestExecGQLWithMockBody
2022-06-22T10:43:22.567-0500, [31mERROR[0m, defra.http, Internal Server Error, {"Error": "no database available", "ErrorVerbose": "no database available\ngithub.com/sourcenetwork/defradb/api/http.dbFromContext\n\t/Users/o/dev/defradb/api/http/handler.go:109\ngithub.com/sourcenetwork/defradb/api/http.getBlockHandler\n\t/Users/o/dev/defradb/api/http/handlerfuncs.go:175\ngithub.com/sourcenetwork/defradb/api/http.(*handler).handle.func1\n\t/Users/o/dev/defradb/api/http/handler.go:74\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2084\ngithub.com/go-chi/chi/v5.(*Mux).routeHTTP\n\t/Users/o/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:442\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2084\ngithub.com/sourcenetwork/defradb/api/http.loggerMiddleware.func1\n\t/Users/o/dev/defradb/api/http/logger.go:57\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2084\ngithub.com/go-chi/chi/v5.(*Mux).ServeHTTP\n\t/Users/o/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:88\ngithub.com/sourcenetwork/defradb/api/http.testRequest\n\t/Users/o/dev/defradb/api/http/handlerfuncs_test.go:690\ngithub.com/sourcenetwork/defradb/api/http.TestGetBlockHandlerWithNoDB\n\t/Users/o/dev/defradb/api/http/handlerfuncs_test.go:553\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1439\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_arm64.s:1263"}
2022-06-22T10:43:22.567-0500, [34mINFO[0m, defra.http, Request, {"Method": "GET", "Path": "/api/v0/blocks/bafybeidembipteezluioakc2zyke4h5fnj4rr3uaougfyxd35u3qzefzhm", "Status": 500, "Length": 112, "Elapsed": 0.00005525}
--- FAIL: TestGetBlockHandlerWithNoDB (0.00s)
    handlerfuncs_test.go:563: 
        	Error Trace:	handlerfuncs_test.go:563
        	Error:      	"" does not contain "no database available"
        	Test:       	TestGetBlockHandlerWithNoDB
```

```
2022-06-22T15:46:51.698Z, ESC[31mERRORESC[0m, defra.http, Internal Server Error, {"Error": "no database available", "ErrorVerbose": "no database available\ngithub.com/sourcenetwork/defradb/api/http.dbFromContext\n\t/home/o/dev/defradb/api/http/handler.go:109\ngithub.com/sourcenetwork/defradb/api/http.getBlockHandler\n\t/home/o/dev/defradb/api/http/handlerfuncs.go:175\ngithub.com/sourcenetwork/defradb/api/http.(*handler).handle.func1\n\t/home/o/dev/defradb/api/http/handler.go:74\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2047\ngithub.com/go-chi/chi/v5.(*Mux).routeHTTP\n\t/home/o/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:442\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2047\ngithub.com/sourcenetwork/defradb/api/http.loggerMiddleware.func1\n\t/home/o/dev/defradb/api/http/logger.go:57\nnet/http.HandlerFunc.ServeHTTP\n\t/usr/local/go/src/net/http/server.go:2047\ngithub.com/go-chi/chi/v5.(*Mux).ServeHTTP\n\t/home/o/go/pkg/mod/github.com/go-chi/chi/v5@v5.0.7/mux.go:88\ngithub.com/sourcenetwork/defradb/api/http.testRequest\n\t/home/o/dev/defradb/api/http/handlerfuncs_test.go:690\ngithub.com/sourcenetwork/defradb/api/http.TestGetBlockHandlerWithDSKeyWithNoDB\n\t/home/o/dev/defradb/api/http/handlerfuncs_test.go:535\ntesting.tRunner\n\t/usr/local/go/src/testing/testing.go:1259\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1581"}
2022-06-22T15:46:51.698Z, ESC[34mINFOESC[0m, defra.http, Request, {"Method": "GET", "Path": "/api/v0/blocks/CIQGIYCQ7GIJSXIQ4AUFVTQUJYP2K2TZDDXIA5IMLROHX3JXBSILSOY", "Status": 500, "Length": 112, "Elapsed": 0.000165912}
--- FAIL: TestGetBlockHandlerWithDSKeyWithNoDB (0.00s)
    handlerfuncs_test.go:545: 
                Error Trace:    handlerfuncs_test.go:545
                Error:          "" does not contain "no database available"
                Test:           TestGetBlockHandlerWithDSKeyWithNoDB
```

```
--- FAIL: TestGetBlockHandlerWithGetBlockstoreError (0.00s)
    handlerfuncs_test.go:584: 
                Error Trace:    handlerfuncs_test.go:584
                Error:          "" does not contain "ipld: could not find bafybeidembipteezluioakc2zyke4h5fnj4rr3uaougfyxd35u3qzefzhm"
                Test:           TestGetBlockHandlerWithGetBlockstoreError
```

```
--- FAIL: TestLoadSchemaHandlerWithAddSchemaError (0.00s)
    handlerfuncs_test.go:466: 
                Error Trace:    handlerfuncs_test.go:466
                Error:          "" does not contain "Syntax Error GraphQL (2:1) Unexpected Name"
                Test:           TestLoadSchemaHandlerWithAddSchemaError
```
```
--- FAIL: TestExecGQLHandlerContentTypeJSONWithJSONError (0.00s)
    handlerfuncs_test.go:255: 
                Error Trace:    handlerfuncs_test.go:255
                Error:          "" does not contain "invalid character"
                Test:           TestExecGQLHandlerContentTypeJSONWithJSONError
```

```
--- FAIL: TestExecGQLHandlerContentTypeJSONWithJSONError (0.00s)
    handlerfuncs_test.go:255: 
                Error Trace:    handlerfuncs_test.go:255
                Error:          "" does not contain "invalid character"
                Test:           TestExecGQLHandlerContentTypeJSONWithJSONError
```

--- a/api/http/errors_test.go
+++ b/api/http/errors_test.go
@@ -21,7 +21,12 @@ import (
 	"github.com/stretchr/testify/assert"
 )
 
+func CleanupEnv() {
+	env = ""
+}
+
 func TestFormatError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
 	env = "prod"
 	s := formatError(errors.New("test error"))
 	assert.Equal(t, "", s)
@@ -33,6 +38,7 @@ func TestFormatError(t *testing.T) {
 }
 
 func TestHandleErrOnBadRequest(t *testing.T) {
+	t.Cleanup(CleanupEnv)
 	env = "dev"
 	f := func(rw http.ResponseWriter, req *http.Request) {
 		handleErr(req.Context(), rw, errors.New("test error"), http.StatusBadRequest)
@@ -65,6 +71,7 @@ func TestHandleErrOnBadRequest(t *testing.T) {
 }
 
 func TestHandleErrOnInternalServerError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
 	env = "dev"
 	f := func(rw http.ResponseWriter, req *http.Request) {
 		handleErr(req.Context(), rw, errors.New("test error"), http.StatusInternalServerError)
@@ -96,6 +103,7 @@ func TestHandleErrOnInternalServerError(t *testing.T) {
 }
 
 func TestHandleErrOnNotFound(t *testing.T) {
+	t.Cleanup(CleanupEnv)
 	env = "dev"
 	f := func(rw http.ResponseWriter, req *http.Request) {
 		handleErr(req.Context(), rw, errors.New("test error"), http.StatusNotFound)
@@ -128,6 +136,7 @@ func TestHandleErrOnNotFound(t *testing.T) {
 }
 
 func TestHandleErrOnDefault(t *testing.T) {
+	t.Cleanup(CleanupEnv)
 	env = "dev"
 	f := func(rw http.ResponseWriter, req *http.Request) {
 		handleErr(req.Context(), rw, errors.New("Unauthorized"), http.StatusUnauthorized)

--- a/api/http/handlerfuncs_test.go
+++ b/api/http/handlerfuncs_test.go
@@ -115,6 +115,8 @@ func TestDumpHandlerWithNoError(t *testing.T) {
 }
 
 func TestDumpHandlerWithDBError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	errResponse := errorResponse{}
 	testRequest(testOptions{
 		Testing:        t,
@@ -132,6 +134,8 @@ func TestDumpHandlerWithDBError(t *testing.T) {
 }
 
 func TestExecGQLWithNilBody(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	errResponse := errorResponse{}
 	testRequest(testOptions{
 		Testing:        t,
@@ -150,6 +154,8 @@ func TestExecGQLWithNilBody(t *testing.T) {
 }
 
 func TestExecGQLWithEmptyBody(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	errResponse := errorResponse{}
 	testRequest(testOptions{
 		Testing:        t,
@@ -177,6 +183,8 @@ func (m *mockReadCloser) Read(p []byte) (n int, err error) {
 }
 
 func TestExecGQLWithMockBody(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	mockReadCloser := mockReadCloser{}
 	// if Read is called, it will return error
 	mockReadCloser.On("Read", mock.AnythingOfType("[]uint8")).Return(0, fmt.Errorf("error reading"))
@@ -199,6 +207,8 @@ func TestExecGQLWithMockBody(t *testing.T) {
 }
 
 func TestExecGQLWithNoDB(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	errResponse := errorResponse{}
 	stmt := `
 mutation {
@@ -224,6 +234,8 @@ mutation {
 }
 
 func TestExecGQLHandlerContentTypeJSONWithJSONError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	// statement with JSON formatting error
 	stmt := `
 [
@@ -302,6 +314,8 @@ func TestExecGQLHandlerContentTypeJSON(t *testing.T) {
 }
 
 func TestExecGQLHandlerContentTypeFormURLEncoded(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	errResponse := errorResponse{}
 	testRequest(testOptions{
 		Testing:        t,
@@ -388,6 +402,8 @@ mutation {
 }
 
 func TestLoadSchemaHandlerWithReadBodyError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	mockReadCloser := mockReadCloser{}
 	// if Read is called, it will return error
 	mockReadCloser.On("Read", mock.AnythingOfType("[]uint8")).Return(0, fmt.Errorf("error reading"))
@@ -410,6 +426,8 @@ func TestLoadSchemaHandlerWithReadBodyError(t *testing.T) {
 }
 
 func TestLoadSchemaHandlerWithoutDB(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	stmt := `
 type user {
 	name: String 
@@ -438,6 +456,8 @@ type user {
 }
 
 func TestLoadSchemaHandlerWithAddSchemaError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	ctx := context.Background()
 	defra := testNewInMemoryDB(t, ctx)
 
@@ -508,6 +528,8 @@ type user {
 }
 
 func TestGetBlockHandlerWithMultihashError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	errResponse := errorResponse{}
 	testRequest(testOptions{
 		Testing:        t,
@@ -524,7 +546,10 @@ func TestGetBlockHandlerWithMultihashError(t *testing.T) {
 	assert.Equal(t, "Bad Request", errResponse.Errors[0].Extensions.HTTPError)
 	assert.Equal(t, "illegal base32 data at input byte 0", errResponse.Errors[0].Message)
 }
+
 func TestGetBlockHandlerWithDSKeyWithNoDB(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	cID, err := cid.Parse("bafybeidembipteezluioakc2zyke4h5fnj4rr3uaougfyxd35u3qzefzhm")
 	if err != nil {
 		t.Fatal(err)
@@ -549,6 +574,8 @@ func TestGetBlockHandlerWithDSKeyWithNoDB(t *testing.T) {
 }
 
 func TestGetBlockHandlerWithNoDB(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	errResponse := errorResponse{}
 	testRequest(testOptions{
 		Testing:        t,
@@ -567,6 +594,8 @@ func TestGetBlockHandlerWithNoDB(t *testing.T) {
 }
 
 func TestGetBlockHandlerWithGetBlockstoreError(t *testing.T) {
+	t.Cleanup(CleanupEnv)
+	env = "dev"
 	ctx := context.Background()
 	defra := testNewInMemoryDB(t, ctx)
 

